gitlabform/configuration/core.py

```diff
@@ -162,13 +163,64 @@ def is_group_skipped(self, group) -> bool:
     def is_skipped(self, an_array: list, item: str) -> bool:
         pass
 
+    @staticmethod
+    def validate_break_inheritance_flag(config, **kwargs):
+        has_invalid_flag = False
+        level = None
+        for key, value in config.items():
+            if "inherit" == key:
+                if kwargs["parent"] == "empty" and (
+                    kwargs["level"] == "group"
+                    or kwargs["level"] == "project"
+                    or kwargs["level"] == "subgroup"
+                ):
+                    level = kwargs["level"]
+                    has_invalid_flag = True
+                elif kwargs["level"] == "common":
+                    level = "common"
+                    has_invalid_flag = True
+            if has_invalid_flag:
+                fatal(
+                    "The inheritance break flag cannot be placed at the {config_level} level\n"
+                    "because {config_level} level is the highest level in the configuration file.\n".format(
+                        config_level=level
+                    ),
+                    exit_code=EXIT_PROCESSING_ERROR,
+                )
+            elif type(value) is CommentedMap:
+                ConfigurationCore.validate_break_inheritance_flag(
+                    value, level=kwargs["level"], parent=kwargs["parent"]
+                )
+
     @staticmethod
     def merge_configs(more_general_config, more_specific_config) -> dict:
         """
         :return: merge more general config with more specific configs.
                  More specific config values take precedence over more general ones.
         """
-        return dict(merge({}, more_general_config, more_specific_config))
+
+        merged_dict = merge({}, more_general_config, more_specific_config)
+
+        def break_inheritance(specific_config, parent_key=""):
+            for key, value in specific_config.items():
+                if "inherit" == key:
```

Comment:
It's not enough to have the key `inherit` - it must also be set to boolean `False`

Reply:
I left it this way on purpose because I was unsure if this is the "inheritance-break flag" that everyone is set on. I did remember there was a bit of a discussion on what we wanted to rename it to. 

However, since I decided to go with what you suggested initially, I will add that check as well.